javassist-maven-plugin/src/main/java/nl/topicus/plugins/maven/javassist/JavassistMojo.java

```diff
@@ -34,245 +35,248 @@
-		for (final String runtimeResource : classpathElements) {
-			URL url = resolveUrl(runtimeResource);
-			if (url != null) {
-				classPath.add(url);
-			}
-		}
-
-		return classPath;
-	}
-
-	private void loadClassPath(final ClassLoader contextClassLoader,
-			final List<URL> urls) {
-		if (urls.size() <= 0)
-			return;
-
-		final URLClassLoader pluginClassLoader = URLClassLoader.newInstance(
-				urls.toArray(new URL[urls.size()]), contextClassLoader);
-		currentThread().setContextClassLoader(pluginClassLoader);
-	}
-
-	private URL resolveUrl(final String resource) {
-		try {
-			return new File(resource).toURI().toURL();
-		} catch (final MalformedURLException e) {
-			throw new RuntimeException(e.getMessage(), e);
-		}
-	}
-
-	@Override
-	public void addMessage(File file, int line, int pos, String message,
-			Throwable e) {
-		buildContext.addMessage(file, line, pos, message,
-				BuildContext.SEVERITY_ERROR, e);
-	}
-
-	@Override
-	public void debug(String message) {
-		getLog().info(message);
-	}
-
-	@Override
-	public void debug(String message, Throwable throwable) {
-		getLog().info(message, throwable);
-	}
-
-	@Override
-	public void info(String message) {
-		getLog().info(message);
-	}
-
-	@Override
-	public void info(String message, Throwable throwable) {
-		getLog().info(message, throwable);
-	}
-
-	@Override
-	public void warn(String message) {
-		getLog().warn(message);
-	}
-
-	@Override
-	public void warn(String message, Throwable throwable) {
-		getLog().warn(message, throwable);
-	}
-
-	@Override
-	public void error(String message) {
-		getLog().error(message);
-	}
-
-	@Override
-	public void error(String message, Throwable throwable) {
-		getLog().error(message, throwable);
-	}
+    private static final Class<ClassTransformer> TRANSFORMER_TYPE = ClassTransformer.class;
+
+    @Component
+    private BuildContext buildContext;
+
+    @Parameter(property = "project", defaultValue = "${project}")
+    private MavenProject project;
+
+    @Parameter(property = "transformerClass", required = true)
+    private String transformerClass;
+
+    @Parameter(property = "processInclusions", required = true)
+    private List<String> processInclusions;
+
+    @Parameter(property = "processExclusions")
+    private List<String> processExclusions;
+
+    @Parameter(property = "exclusions")
+    private List<String> exclusions;
+
+    @Parameter(property = "outputDirectory", defaultValue = "${project.build.outputDirectory}")
+    private String outputDirectory;
+
+    public void execute() throws MojoExecutionException {
+        final ClassLoader originalContextClassLoader = currentThread()
+            .getContextClassLoader();
+        try {
+            final List<String> classpathElements = getCompileClasspathElements();
+            loadClassPath(originalContextClassLoader,
+                generateClassPathUrls(classpathElements));
+            transform(Arrays.asList(project.getBuild().getOutputDirectory()));
```

Comment:
I assume you have a "inline setup" in which you define maven-javassist-plugin in the project directly in order to replace the compiled classes with the altered version.

Your change does not take into account that the maven-javassist-plugin can also be used in a separate project so the altered classes are bundled in a separate jar. (See README.md)